File: 55-jump-game/55-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int n = nums.size();
        if(n == 0) return 0;
        if(n == 1) return true;
        if(nums[0] == 0) return 0;
        
        int maxJump = nums[0];
        
        for(int i=1;i<n;i++){
            
            if(i > maxJump){
                return false;
            }
            maxJump = max(maxJump,i + nums[i]);            
        }
        return true;
    }
};

File: 55-jump-game/55-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int max_jump = 0;
        for(int i=0;i<nums.length;i++){
            if(max_jump < i) return false;
            max_jump = Math.max(max_jump,i + nums[i]);
        }
        return true;
    }
}